pkg/util/aad/aad.go

```diff
@@ -95,3 +99,44 @@ func GetToken(ctx context.Context, log *logrus.Entry, oc *api.OpenShiftCluster,
 
 	return token, nil
 }
+
+// GetClusterSPObjectID returns a service principal objectID for a cluster.
+// It retries in case of Authorization_IdentityNotFound error as it is likely due to AAD sync delay.
+func GetClusterSPObjectID(ctx context.Context, log *logrus.Entry, oc *api.OpenShiftCluster, token *adal.ServicePrincipalToken) (string, error) {
+	spp := &oc.Properties.ServicePrincipalProfile
+
+	spGraphAuthorizer := autorest.NewBearerAuthorizer(token)
+	applications := graphrbac.NewApplicationsClient(spp.TenantID, spGraphAuthorizer)
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 15*time.Minute)
```

Comment:
Not sure what is the reasonable timeout here. I vaguely remember someone mentioning that sync/propogation can take up to 15 minutes.

Reply:
Decided to set to 2 minutes for now